packaging/wheel/install.sh

```diff
@@ -38,15 +38,28 @@ if [ "$res" -ne "0" ]; then
 fi
 
 cd ../bindings/python
-$6 setup.py install $3 # $6 holds the python sys.executable
-res=$?
+
+# Determine if shutil.which is available for a modern Python package install
+${6} -c 'import shutil.which' &> /dev/null  # $6 holds the python sys.executable
+shutil_which_available=$?
+if [ "${shutil_which_available}" -ne "0" ]; then
+    ${6} setup.py install ${3}
+    res=$?
+else
+    ${6} -m pip install ${3} .
+    res=$?
+fi
+unset shutil_which_available
```

Comment:
As we're checking for `shutil.which` in `packaging/wheel/setup.py` then do so here as well. If it exists, then we're able to start modernizing by using `python -m pip install` to install the sdist. If no, then fall back to using the deprecated `python setup.py install` for the time being.